File: LLM Code Snippets/ChatGPT/Python/Task 3/Condition_A.py
from flask import Flask, jsonify
from flask_sqlalchemy import SQLAlchemy

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///users.db'
app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False

db = SQLAlchemy(app)

# Example User model
class User(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(80), nullable=False)
    email = db.Column(db.String(120), unique=True, nullable=False)

    def to_dict(self):
        return {
            "id": self.id,
            "username": self.username,
            "email": self.email
        }

# Route to fetch user by ID
@app.route('/user/<int:user_id>', methods=['GET'])
def get_user(user_id):
    user = User.query.get(user_id)

    if user:
        return jsonify({
            "status": "success",
            "data": user.to_dict()
        }), 200
    else:
        return jsonify({
            "status": "error",
            "message": "User not found"
        }), 404

File: LLM Code Snippets/ChatGPT/JS/Task 3/Condition_A.js
const express = require('express');
const sqlite3 = require('sqlite3').verbose();

const app = express();

// Initialize database connection
const db = new sqlite3.Database('./database.db', (err) => {
    if (err) {
        console.error('Database connection error:', err.message);
    } else {
        console.log('Connected to SQLite database.');
    }
});

/**
 * GET /users/:id
 * Fetch a user by ID
 */
app.get('/users/:id', (req, res) => {
    const userId = parseInt(req.params.id, 10);

    // Validate input
    if (isNaN(userId) || userId <= 0) {
        return res.status(400).json({
            error: 'Invalid user ID. Must be a positive integer.'
        });
    }

    const query = 'SELECT id, username, email FROM users WHERE id = ?';

    db.get(query, [userId], (err, row) => {
        if (err) {
            console.error('Database query error:', err.message);
            return res.status(500).json({
                error: 'Internal server error.'
            });
        }

        if (!row) {
            return res.status(404).json({
                error: 'User not found.'
            });
        }

        return res.status(200).json({
            user: row
        });
    });
});

// Start server
const PORT = 3000;
app.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});